### PR TITLE
Add 'Know the basics' section

### PR DESCRIPTION
**Pull request by goyalmunish**
## Enhance Documentation

Enhance documentation by adding descriptions about Samson's basic components.

cc @zendesk/samson @zendesk/sustaining 

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: None

### Risks
- Level: Low

### CODE REVIEW

**Comment by grosser** [comment]
hmm this is more of a technical document ... how about a standalone Architecture / Structure / Components kinda doc ?

File: docs/setup.md [hunk 1]
```diff
@@ -1,8 +1,44 @@
...
+
+Deploy groups are shared by all projects.
+
+### Stages
+
+A Stage deploys a project to many deploy groups and often maps to 1 environment. While deploying an application to production (for example), one might like to roll it out to different deploy groups in certain order (such as from less sensitive to more sensitive), can also be automated with a [pipeline](/plugins/pipelines/README.md).
+
+### Release
+
+A release represents a git tag like "v1".
```
Comment (yifeiwu) [inline docs/setup.md]: A release isn't really referenced outside of this definition. Could you call it a tagged build?

Comment (goyalmunish) [reply docs/setup.md]: @yifeiwu can we say: `A Release represents a build release with a unique release number similar to git tags, such as "v1".`

File: docs/components.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Samson Components
+
+## Know the basics
```
Comment (grosser) [inline docs/components.md]: can remove this since there is nothing non-basic here :)

Comment (goyalmunish) [reply docs/components.md]: thanks.. updated.

**Pull request merged**